nestjs/src/certificates/certificates.service.ts

```diff
@@ -82,4 +83,26 @@ export class CertificationsService {
       },
     };
   }
+
+  public async removeCertificate(studentId: number) {
+    const certificate = await this.certificateRepository.findOne({
+      where: { studentId },
+    });
+
+    if (!certificate) {
+      throw new NotFoundException('Certificate not found');
+    }
+
+    await Promise.all([
+      new Promise((resolve, reject) => {
+        this.httpService
+          .delete(`${this.configService.awsServices.restApiUrl}/certificate/${certificate.s3Key}`)
+          .subscribe({
+            next: response => resolve(response),
+            error: error => reject(error),
+          });
+      }),
```

Comment:
damn, my bad: I forgot that this thing returns Observable by default, your previous code with `lastValueOf` was correct, please use it instead of this subscription